spec/services/process_payment_intent_spec.rb

```diff
@@ -32,13 +32,15 @@
 
       before do
         allow(order).to receive(:deliver_order_confirmation_email)
+        payment.adjustment.update_attribute(:eligible, false)
       end
 
       it "completes the payment" do
         service.call!
         payment.reload
         expect(payment.state).to eq("completed")
         expect(payment.cvv_response_message).to be nil
+        expect(payment.adjustment.eligible).to be true
```

Comment:
I was originally doing 
`expect(payment).to have_received(:reinstate_adjustment_eligibility!)`
but it wasn't working as expected - it said it wasn't receiving the message, even though it was. 

So I went with this approach, which is a more useful test, although it does require reaching into the adjustment implementation just a little bit.